main/acle.md

```diff
@@ -9546,25 +9680,2141 @@ possible to write these operations using normal C arithmetic. For example:
+
+``` c
+  // Variants are also available for _s8_x2, _u16_x2, _s16_x2, _f16_x2,
+  // _bf16_x2, _u32_x2, _s32_x2, _f32_x2, _u64_x2, _s64_x2 and _f64_x2
+  __attribute__((arm_streaming))
+  svuint8x2_t svsel[_u8_x2](svcount_t png, svuint8x2_t zn, svuint8x2_t zm);
+  
+  
+  // Variants are also available for _s8_x4, _u16_x4, _s16_x4, _f16_x4,
+  // _bf16_x4, _u32_x4, _s32_x4, _f32_x4, _u64_x4, _s64_x4 and _f64_x4
+  __attribute__((arm_streaming))
+  svuint8x4_t svsel[_u8_x4](svcount_t png, svuint8x4_t zn, svuint8x4_t zm);
+  
+  ```
+
+#### URSHL, SRSHL (single)
+
+Multi-vector rounding shift left
+
+``` c
+  // Variants are also available for _single_u8_x2, _single_u16_x2,
+  // _single_s16_x2, _single_u32_x2, _single_s32_x2, _single_u64_x2
+  // and _single_s64_x2
+  __attribute__((arm_streaming))
+  svint8x2_t svrshl[_single_s8_x2](svint8x2_t zdn, svint8_t zm);
+  
+  
+  // Variants are also available for _single_u8_x4, _single_u16_x4,
+  // _single_s16_x4, _single_u32_x4, _single_s32_x4, _single_u64_x4
+  // and _single_s64_x4
+  __attribute__((arm_streaming))
+  svint8x4_t svrshl[_single_s8_x4](svint8x4_t zdn, svint8_t zm);
+  
+  ```
+
+#### URSHL, SRSHL (multi)
+
+Multi-vector rounding shift left
+
+``` c
+  // Variants are also available for _u8_x2, _u16_x2, _s16_x2, _u32_x2, _s32_x2,
+  // _u64_x2 and _s64_x2
+  __attribute__((arm_streaming))
+  svint8x2_t svrshl[_s8_x2](svint8x2_t zdn, svint8x2_t zm);
+  
+  
+  // Variants are also available for _u8_x4, _u16_x4, _s16_x4, _u32_x4, _s32_x4,
+  // _u64_x4 and _s64_x4
+  __attribute__((arm_streaming))
+  svint8x4_t svrshl[_s8_x4](svint8x4_t zdn, svint8x4_t zm);
+  
+  ```
+
+#### SQRSHR, UQRSHR
+
+Multi-vector saturating rounding shift right narrow
+
+``` c
+  // Variants are also available for _u16_x2
+  __attribute__((arm_streaming))
+  svint16_t svqrshr[_s16_x2](svint32x2_t zn, uint64_t imm);
+  
+
+  // Variants are also available for _u8_x4
+  __attribute__((arm_streaming))
+  svint8_t svqrshr[_s8_x4](svint32x4_t zn, uint64_t imm);
+  
+
+  // Variants are also available for _u16_x4
+  __attribute__((arm_streaming))
+  svint16_t svqrshr[_s16_x4](svint64x4_t zn, uint64_t imm);
+  
+  
+  ```
+
+#### SQRSHRN, UQRSHRN
+
+Multi-vector saturating rounding shift right narrow and interleave
+
+``` c
+  // Variants are also available for _u16_x2
+  __attribute__((arm_streaming_compatible))
+  svint16_t svqrshrn[_s16_x2](svint32x2_t zn, uint64_t imm);
+  
+  
+  // Variants are also available for _u8_x4
+  __attribute__((arm_streaming))
+  svint8_t svqrshrn[_s8_x4](svint32x4_t zn, uint64_t imm);
+
+
+  // Variants are also available for _u16_x4
+  __attribute__((arm_streaming))
+  svint16_t svqrshrn[_s16_x4](svint64x4_t zn, uint64_t imm);
+  
+  ```
+
+#### SQRSHRU
+
+Multi-vector saturating rounding shift right unsigned narrow
+
+``` c
+  
+  __attribute__((arm_streaming))
+  svuint16_t svsqrshru[_u16_x2](svint32x2_t zn, uint64_t imm);
```

Comment:
Not saying this is obviously the best choice, but the convention in `arm_neon.h` is to:
* drop the `s`
* use a signed suffix rather than unsigned suffix, reflecting the source type rather than the destination type

SVE2 followed the same approach, for consistency.  So I think this should be `svqrshru[_s32_x2]`.